shuup/admin/forms/fields.py

```diff
@@ -165,3 +183,51 @@ def __init__(self, choices=(), required=True, widget=None, label=None, initial=N
 
     def clean(self, value):
         return ",".join(super(WeekdayField, self).clean(value))
+
+
+class ObjectSelect2ModelField(Select2ModelField):
+    """
+    Class for select2 form fields.
+    Replacement for the class Select2ModelField.
+    """
+
+    def widget_attrs(self, widget):
+        attrs = super(ObjectSelect2ModelField, self).widget_attrs(widget)
+        attrs["class"] = "object-selector"
```

Comment:
I think you can write new classes to handle the new objects. Instead of received a model as arguments on __init__. This will allow us to write any kind of object selector not relying on a model directly, for example:

```py
random_objects = ObjectSelect2MultipleField(
        label=_("Random Objects"),
        help_text=_("Set will retrieve random objects."),
        selector="myapp.anything-i-want",
        required=False,
)
```

and you can write some utils to convert a model into a selector:
```py
categories = ObjectSelect2MultipleField(
        label=_("Categories"),
        selector=get_selector_for_model(Category),
        required=False,
)
```

this `get_selector_for_model` can be:

```py
def get_selector_for_model(model):
    return f"{model._meta.app_label}.{model._meta.model_name}"
```

Reply:
Currently it is relying on the model to convert the data in the methods `prepare_value` and `to_python`. So we still need the model. If the selector is not a model we need a way of indicating how to convert the data (maybe adding a couple of parameters for passing a function). I'd like to discuss it and see some use cases to understand better the idea.